modules/eclickadsBidAdapter.js

```diff
@@ -0,0 +1,98 @@
+import { NATIVE } from '../src/mediaTypes.js';
+import { convertOrtbRequestToProprietaryNative } from '../src/native.js';
+import { getWindowTop } from '../src/utils.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+
+// ***** ECLICKADS ADAPTER *****
+export const BIDDER_CODE = 'eclickads';
+const DEFAULT_CURRENCY = ['USD'];
+const DEFAULT_TTL = 1000;
+export const ENDPOINT = 'https://g.eclick.vn/rtb_hb_request?fosp_uid=';
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [NATIVE],
+  isBidRequestValid: (bid) => {
+    return !!bid && !!bid.params && !!bid.bidder && !!bid.params.zid;
+  },
+  buildRequests: (validBidRequests = [], bidderRequest) => {
+    validBidRequests = convertOrtbRequestToProprietaryNative(validBidRequests);
+
+    const ortb2ConfigFPD = bidderRequest.ortb2.site.ext?.data || {};
+    const ortb2Device = bidderRequest.ortb2.device;
+    const ortb2Site = bidderRequest.ortb2.site;
+
+    const winTop = getWindowTop();
+    const device = getDevice(ortb2Device.ua);
```

Comment:
There's potentially a rich device object available, why not get to whole thing?

Reply:
Initially, we intended to take several parameters, not the whole device object.
But that's a good thing. Modified to get more(all) params from device object.